docs/providers/tutorial-create-a-provider.md

```diff
@@ -0,0 +1,754 @@
+- Still necessary for the provider to function correctly
+
+By that criterion, authentication values are very good configuration values. Our API does not
+handle authentication though, so we won't need it here.
+
+#### Testing a SDK for serverless WebAssembly compilation
+
+Sometimes the third party resource you want to integrate provides a Rust SDK.
+But beware! There are a few libraries that will fail (either at compile time or
+at run time) when compiled for serverless WASM. So if you are not sure that the
+SDK will work with serverless wasm, it is better to find a classic API (on top
+of HTTP or WebSocket protocols) documentation to communicate with the third
+party resource. We will cover "finding out if the SDK works in serverless wasm"
+when setting up the project.
+
+When things don't work as expected, you can ask for help or guidance online[^serverlesswasm].
+
+[^serverlesswasm]: The space of serverless WebAssembly is still
+    young and a lot of package maintainers can make the change or help you make the change to get a library to compile and
+    run for WebAssembly. Sometimes it means finding out support is [already there](https://github.com/gyscos/zstd-rs/pull/139),
+    sometimes it means [adding items to their roadmap](https://github.com/awslabs/aws-sdk-rust/discussions/679)
+
+#### Checkpoint
+
+At that point you must know
+- what configuration values are necessary to communicate with the third party
+  resource
+- a documentation/reference to be able to programmatically communicate with said
+  resources
+
+### Clone the Providers repo
+
+The current Fiberplane [providers repo](https://github.com/fiberplane/providers)
+contains a "sample"
+[provider](https://github.com/fiberplane/providers/tree/main/providers/sample)
+that is a good base to make your provider from: our CI pipeline guarantees that
+the sample provider is always up-to-date with our supporting libraries, so it is
+the quickest way to obtain a starting template.
+
+Clone the repository and keep the location of the sample provider somewhere handy:
+```console
+$ git clone https://github.com/fiberplane/providers
+$ export SAMPLE_PROVIDER_PATH="$(pwd)/providers/providers/sample"
+```
+
+#### Checkpoint
+
+If you never restart your terminal session again, we will have a valid path for
+`${SAMPLE_PROVIDER_PATH}`. We will not use it for long, but the point is that
+you should have a fresh copy of the Fiberplane/providers repository.
+
+
+
+## Instantiate a Daemon and side-load your sample provider for use in Studio
+
+### Create your new provider repo, and test sample compilation
+
+It's time to create your provider repo. To avoid the usual generic names, we
+will create a `catnip` provider.
+
+```console
+$ mkdir catnip_provider
+$ cd catnip_provider
+$ cp -r "${SAMPLE_PROVIDER_PATH}"/* .
+$ git init .
+```
+
+Once the repo is created, use the Cargo configuration feature to make the
+compiler default to Web Assembly compilation. Create a `.cargo` folder at the
```

Comment:
it might be good to distinguish this from the root `Cargo.toml` here

Reply:
Sure, I'll add a small folder layout to show